presidio-analyzer/analyzer/analysis_explanation.py

```diff
@@ -0,0 +1,40 @@
+class AnalysisExplanation:
+    """ AnalysisExplanation is a class that holds tracing information
+     to explain why PII entities where indentified as such
+    """
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, recognizer, pattern_name,
+                 pattern, original_score, textual_explanation=None):
+        """
+        """
+        self.recognizer = recognizer
+        self.pattern_name = pattern_name
+        self.pattern = pattern
+        self.original_score = original_score
+        self.score = original_score
+        self.textual_explanation = textual_explanation
+        self.score_context_improvement = 0
+        self.supportive_context_word = ''
+
```

Comment:
```suggestion
self.validation_result = validation_result
```
We are providing a generic explanation object for pattern recognizers. Validation (checksum) is done automatically, whether the user implemented the validation logic or not. That's why the validation result should also be part of the explanation output.